src/Functions/toDayOfWeek.cpp

```diff
@@ -7,14 +7,38 @@
 namespace DB
 {
 
+namespace
+{
+
+/// Sunday is 1, Monday is 2, and Saturday is 7
+struct DayOfWeekTransformer
+{
+    static constexpr auto name = "DAYOFWEEK";
+
+    static inline UInt8 transform(UInt8 src) { return src % 7 + 1; }
+};
+
+/// Monday is 0, Tuesday is 2, and Sunday is 6
+struct WeekDayTransformer
+{
+    static constexpr auto name = "WEEKDAY";
+
+    static inline UInt8 transform(UInt8 src) { return src - 1; }
+};
+
+}
+
 using FunctionToDayOfWeek = FunctionDateOrDateTimeToSomething<DataTypeUInt8, ToDayOfWeekImpl>;
+using FunctionDayOfWeek = FunctionDateOrDateTimeToSomething<DataTypeUInt8, ToDayOfWeekWithTransformImpl<DayOfWeekTransformer>>;
+using FunctionWeekDay = FunctionDateOrDateTimeToSomething<DataTypeUInt8, ToDayOfWeekWithTransformImpl<WeekDayTransformer>>;
```

Comment:
I like adding a `WEEKDAY` function but it should ideally map internally to `ToDayOfWeek` with mode = 1.